src/base_decorator.rb

```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def intialize(nameable)
+    @nameable = nameable
```

Comment:
- [ ] I think it would be great if you add setter and getter to the namable here so you can add it as `attr_accessor` to be able to access it inside the class as well 💯🤩

You could add it like this 🚀
> `attr_accessor :nameable`